src/utils/getUrlByRequestOptions.ts

```diff
@@ -166,7 +158,7 @@ export function getUrlByRequestOptions(options: ResolvedRequestOptions): URL {
     : ''
   logger.info('auth string:', authString)
 
-  const url = new URL(`${protocol}//${hostname}${path}`)
+  const url = new URL(`${protocol}//${host}${path}`)
```

Comment:
I think we need to add this:

```
url.port = port
```

The port may not be a part of the `options.host` if we trust this test:

https://github.com/mswjs/interceptors/blob/2e98a714bc4364fb201313c1bcd326b3b2767a61/src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.test.ts#L313-L315

There's all sorts of shenanigans in how people construct requests but I trust that test suite had a real use case behind it so I'd keep it.

Reply:
This will also fix the currently failing tests.

Reply:
What is the priority? 

```js
 port: '50176', 
 path: '/resource', 
 host: '127.0.0.1:8080', 
```

What's the port of this request?

Reply:
> What is the priority?

I'd say `options.port` is the priority. We are threading on broken userland here but I think if port has been set explicitly we must always respect it. 

> Maybe we should do:

Yes, I think you're right!